icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -153,8 +159,8 @@ static bool isDigit(UChar32 c) { return inRange(c, 0x0030, 0x0039); }
 
 static bool isNameStart(UChar32 c) {
     return isAlpha(c) || c == UNDERSCORE || inRange(c, 0x00C0, 0x00D6) || inRange(c, 0x00D8, 0x00F6) ||
-           inRange(c, 0x00F8, 0x02FF) || inRange(c, 0x0370, 0x037D) || inRange(c, 0x037F, 0x1FFF) ||
-           inRange(c, 0x200C, 0x200D) || inRange(c, 0x2070, 0x218F) || inRange(c, 0x2C00, 0x2FEF) ||
+           inRange(c, 0x00F8, 0x02FF) || inRange(c, 0x0370, 0x037D) || inRange(c, 0x037F, 0x061B) ||
+           inRange(c, 0x061D, 0x200D) || inRange(c, 0x2070, 0x218F) || inRange(c, 0x2C00, 0x2FEF) ||
```

Comment:
@catamorphism  i see ALM is not namestart, but this change makes U+2000…U+200B isNameStart true.  they are dashes and spaces, and not `ID_Start`

Reply:
BTW, I think this would be far more reliable in using a UnicodeSet. That can be created as the C++ equivalent of a static final immutable object.

Reply:
@srl295 Fixed

@macchiati Done in [780a947](https://github.com/unicode-org/icu/pull/3236/commits/780a947d1a44d6705a2948ed0ff77504af17dc40)